vendors/espressif/boards/ports/ble/nimble/iot_ble_hal_gap.c

```diff
@@ -743,9 +743,14 @@ BTStatus_t prvBTSetAdvData( uint8_t ucAdapterIf,
         xESPStatus = ble_gap_adv_set_fields( &fields );
     }
 
-    if( xESPStatus != 0 )
+    switch( xESPStatus )
     {
-        xStatus = eBTStatusFail;
+        case BLE_HS_EMSGSIZE:
+            xStatus = eBTStatusNoMem;
+            break;
+
+        default:
+            xStatus = eBTStatusFail;
```

Comment:
missing a break statement here.

Reply:
it's the last case, so break is extraneous